client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SparkUtils.java

```diff
@@ -386,9 +402,17 @@ public static synchronized boolean taskAnotherAttemptRunningOrSuccessful(long ta
         int taskIndex = taskInfo.index();
         for (TaskInfo ti : taskAttempts) {
           if (ti.taskId() != taskId) {
-            if (ti.successful()) {
+            if (reportedStageTaskIds.contains(ti.taskId())) {
+              LOG.info(
+                  "StageId={} index={} taskId={} attempt={} another attempt {} has reported shuffle fetch failure, ignore it.",
+                  stageId,
+                  taskIndex,
+                  taskId,
+                  taskInfo.attemptNumber(),
+                  ti.attemptNumber());
```

Comment:
I have a offline discussion with Shuang.

To prevent the race condition that the task two attempts report the shuffle fetch failure concurrently and both true returned due these attempts are both alive.

We will record the reported shuffle fetch failure taskIds in the stage.

Then exclude the TaskInfos that already reported shuffle fetch failure.

Thanks @RexXiong , also cc @pan3793